src/app/preferences/general-preferences/general-preferences.service.ts

```diff
@@ -0,0 +1,17 @@
+namespace Core {
+    export class GeneralPreferencesService {
+      constructor(private $window: ng.IWindowService) {
+        'ngInject';
+        this.$window.localStorage.setItem('current_state', "ON");
+      }
+      getVerticalnavstate(): string {
+        return this.$window.localStorage.getItem('current_state');
+  
+      }
+      setVerticalnavstate(current_state): void {
```

Comment:
-> `setDefaultVerticalNavState(state: VerticalNavState)`

And why don't we define typescript enum for the values:
https://www.typescriptlang.org/docs/handbook/enums.html